### PR TITLE
Fixes to Webhook & RecrdSearch integration

### LINKED ISSUE #4299: Error messages in workflow `getUser` and `getSystem` are wrong
https://github.com/opencrvs/opencrvs-core/blob/692f486028bca99111a857289fd305bae00c74af/packages/workflow/src/features/user/utils.ts#L33
The error messages in `getUser` and `getSystem` aren't actually related to mobile numbers. Let's rewrite the error to be "Unable to retrieve user" and "Unable to retrieve system".

### PR DESCRIPTION
**Pull request by euanmillar**
Farajaland PR: https://github.com/opencrvs/opencrvs-farajaland/pull/441

Closes #4299

### CODE REVIEW

File: packages/user-mgnt/src/features/system/handler.ts [hunk 1]
```diff
@@ -395,22 +399,17 @@ export const getSystemResponseSchema = Joi.object({
   scope: Joi.array().items(Joi.string()),
   sha_secret: Joi.string(),
   practitionerId: Joi.string(),
+  type: Joi.string(),
   settings: Joi.object({
-    dailyQuota: Joi.number()
+    dailyQuota: Joi.number(),
+    webhook: Joi.any().optional()
```
Comment (euanmillar) [inline packages/user-mgnt/src/features/system/handler.ts:405]: hapi-joi seems unable to validate the following which should work:

```
settings: Joi.object({
    dailyQuota: Joi.number(),
    webhook: Joi.array()
      .items(
        Joi.object({
          event: Joi.string().required(),
          permissions: Joi.array().items(Joi.string()).required(),
          _id: Joi.string().optional()
        })
      )
      .optional()
  })
```

File: packages/gateway/src/graphql/config.ts [hunk 2]
```diff
@@ -104,26 +103,28 @@ export const getApolloConfig = (): Config => {
...
-          { userId },
-          { Authorization: request.headers.authorization }
+        const tokenPayload = getTokenPayload(
+          request.headers.authorization.split(' ')[1]
         )
-        if (!user) {
+        const userId = tokenPayload.sub
+        let user: IUserModelData | ISystemModelData
+        const isSystemUser = tokenPayload.scope.indexOf('recordsearch') > -1
+        if (isSystemUser) {
```
Comment (Zangetsu101) [inline packages/gateway/src/graphql/config.ts:112]: @euanmillar event notifications will also go through gateway so we need check for that scope as well

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/gateway/src/graphql/config.ts (hunk 2 above)
Comment (euanmillar) [reply packages/gateway/src/graphql/config.ts:112]: @Zangetsu101 I think that they are fine as they do not go through Apollo

Comment (Zangetsu101) [reply packages/gateway/src/graphql/config.ts:112]: Okay yeah you're right

**Pull request merged**